tests/functional/smoke/droptomb2.sh

```diff
@@ -13,39 +13,45 @@ kvdb_create
 
 keys=10
 
-kvs=$(kvs_create smoke-0 fanout=2) || $?
```

Comment:
why was this removed? I look at other smoke tests and they have the same pattern, but I can't remember why the pattern exists. ShellCheck is giving me this locally (probably because I have a newer version or something, not sure)

```
This function is invoked in an || condition so set -e will be disabled. Invoke separately if failures should cause the script to exit.
```

That is with or without the `|| $?`

https://www.shellcheck.net/wiki/SC2310

Reply:
I don't recall doing that..  Alex?  Before Alex's changes I set it to fanout=2 so that it would have fewer nodes to place keys in, which worked, but the real problem I think was the broken route map find, which is fixed now.  So I don't know why this was whacked...

Reply:
@alexttx

Reply:
`command foo bar || $?` is invalid bash code.  It should be `command foo bar || exit $?`.

That being said, it is not needed for kvs_create because it will exit if it fails.

Reply:
More info...

There are two reasons I use `||`.  One reason is to force exit if a command fails:
```
ls nosuchfile || exit 1
```
The other is to disable the effect of `set -e` for a particular command:
```
ls nosuchfile || true
```

The edit I made was an example of the former, which isn't needed bc the `kvs_create` function will exit if it fails.

Reply:
Function `cmd` exits if the command fails.  I actually tested it by adding an invalid arg to `kvs create`.

Reply:
Good catch. If you have the willpower, I would just remove this pattern from all the smoke tests, and maybe the CLI tests if it exists there.

Reply:
I'll do in a follow-on PR